lib/IDE/Refactoring.cpp

```diff
@@ -3887,7 +3889,1442 @@ bool RefactoringActionConvertToComputedProperty::performChange() {
+        addDo();
+        addAwaitCall(CE, ArgList.ref(), Blocks.SuccessBlock, SuccessParams,
+                     /*HasError=*/HandlerDesc.HasError,
+                     /*AddDeclarations=*/!HandlerDesc.HasError);
+        addFallbackCatch(ErrParam);
+        OS << "\n";
+        convertNodes(CallbackBody);
+
+        clearParams(CallbackParams->getArray());
+        return;
+      }
+    }
+
+    bool RequireDo = !Blocks.ErrorBlock.nodes().empty();
+    // Check if we *actually* need a do/catch (see class comment)
+    if (Blocks.ErrorBlock.nodes().size() == 1) {
+      auto Node = Blocks.ErrorBlock.nodes()[0];
+      if (auto *HandlerCall = TopHandler.isCall(Node)) {
+        auto Res = TopHandler.parse(HandlerCall);
+        if (Res.args().size() == 1) {
+          // Skip if we have the param itself or the name it's bound to
+          auto *SingleDecl = Res.args()[0]->getReferencedDecl().getDecl();
+          auto ErrName = Blocks.ErrorBlock.boundName(ErrParam);
+          RequireDo = SingleDecl != ErrParam &&
+                      !(Res.isError() && SingleDecl &&
+                        SingleDecl->getName().isSimpleName(ErrName));
+        }
+      }
+    }
+
+    if (RequireDo) {
+      addDo();
+    }
+
+    setNames(Blocks.SuccessBlock, SuccessParams);
+    addAwaitCall(CE, ArgList.ref(), Blocks.SuccessBlock, SuccessParams,
+                 /*HasError=*/HandlerDesc.HasError,
+                 /*AddDeclarations=*/true);
+
+    prepareNamesForBody(HandlerDesc.Type, SuccessParams, ErrParams);
+    convertNodes(Blocks.SuccessBlock.nodes());
+
+    if (RequireDo) {
+      clearParams(SuccessParams);
+      // Always use the ErrParam name if none is bound
+      setNames(Blocks.ErrorBlock, ErrParams,
+               HandlerDesc.Type != HandlerResultType::RESULT);
+      addCatch(ErrParam);
+
+      prepareNamesForBody(HandlerDesc.Type, ErrParams, SuccessParams);
+      addCatchBody(ErrParam, Blocks.ErrorBlock);
+    }
+
+    clearParams(CallbackParams->getArray());
+  }
+
+  void addAwaitCall(const CallExpr *CE, ArrayRef<Expr *> Args,
+                    const ClassifiedBlock &SuccessBlock,
+                    ArrayRef<const ParamDecl *> SuccessParams, bool HasError,
+                    bool AddDeclarations) {
+    if (!SuccessParams.empty()) {
+      if (AddDeclarations) {
+        if (SuccessBlock.allLet()) {
+          OS << tok::kw_let;
+        } else {
+          OS << tok::kw_var;
+        }
+        OS << " ";
+      }
+      if (SuccessParams.size() > 1)
+        OS << tok::l_paren;
+      OS << newNameFor(SuccessParams.front());
+      for (const auto Param : SuccessParams.drop_front()) {
+        OS << tok::comma << " ";
+        OS << newNameFor(Param);
+      }
+      if (SuccessParams.size() > 1) {
+        OS << tok::r_paren;
+      }
+      OS << " " << tok::equal << " ";
+    }
+
+    if (HasError) {
+      OS << tok::kw_try << " ";
+    }
+    // TODO: Add 'await' keyword
```

Comment:
Was this just left in accidentally?

Reply:
Yeah, thanks. It isn't actually a keyword, so "await" will be fine.